guides/creating_a_new_component.md

```diff
@@ -0,0 +1,386 @@
+			loading_status,
+			show_label: true,
+			mode: "dynamic",
+			value: "#000000",
+			label: "ColorPicker"
+		});
+
+		const item: HTMLInputElement = getByDisplayValue("#000000");
+		assert.equal(item.value, "#000000");
+	});
+
+	test("changing the color should update the value", async () => {
+		const { component, getByDisplayValue } = render(ColorPicker, {
+			loading_status,
+			show_label: true,
+			mode: "dynamic",
+			value: "#000000",
+			label: "ColorPicker"
+		});
+
+		const item: HTMLInputElement = getByDisplayValue("#000000");
+
+		assert.equal(item.value, "#000000");
+
+		await component.$set({
+			value: "#FFFFFF"
+		});
+
+		assert.equal(component.value, "#FFFFFF");
+	});
+});
+```
+
+the third one is the index.ts file:
+
+```typescript
+export { default as Component } from "./ColorPicker.svelte";
+export const modes = ["static", "dynamic"];
+```
+
+- Add the mapping for your component in the [directory.ts file](https://github.com/gradio-app/gradio/blob/main/ui/packages/app/src/components/directory.ts). To do this, copy and paste the mapping line of any component and edit its text. The key name must be the lowercase version of the actual component name in the Python library. So for example, for the ColorPicker component the mapping looks like this: 
+
+```typescript
+export const component_map = {
+...
+colorpicker: () => import("./ColorPicker"),
+...
+}
+```
+
+### Step 2.1 . Writing Unit Test for Svelte Component
+
+When developing new components, you should also write a suite of unit tests for it. The tests should be placed in the new component's folder in a file named MyAwesomeComponent.test.ts. Again, as above, take a cue from the tests of other components (e.g. [Textbox.test.ts](https://github.com/gradio-app/gradio/blob/main/ui/packages/app/src/components/Textbox/Textbox.test.ts)) and add as many unit tests as you think are appropriate to test all the different aspects and functionalities of the new component.
+
+### Step 3 - Create a New Demo
+
+The last step is to create a demo in the [gradio/demo folder](https://github.com/gradio-app/gradio/tree/main/demo), which will use the newly added component. Again, the suggestion is to reference an existing demo. Write the code for the demo in a file called run.py, add the necessary requirements and an image showing the application interface. Finally add a gif showing its usage. 
+You can take a look at the [demo](https://github.com/gradio-app/gradio/tree/main/demo/color_picker) created for the ColorPicker, where an icon and a color selected through the new component is taken as input, and the same icon colored with the selected color is returned as output.
+
+To test the application:
+
+- run on a terminal `python path/demo/run.py` which starts the backend at the address [http://localhost:7860](http://localhost:7860);
+- in another terminal, from the ui folder, run `pnpm dev` to start the frontend at [localhost:3000](localhost:3000) with hot reload functionalities.
```

Comment:
I think this fine for now but we should probably revisit this when we have decided exactly what we want to do with demos and how we want to separate them.